src/plugins/intel_gpu/src/graph/program_helpers.cpp

```diff
@@ -92,6 +92,36 @@ void onednn_add_fusing_helpers::for_eltwise(
     }
 }
 
+static bool is_direct_ancestor(const program_node& child, const program_node& target) {
+    // is_direct_ancestor function is added to detect pattern like below.
+    // It is necessary to use onednn sum post operation in such case for better performance.
+    // In such case, 'A' can have two connections.
+    //   ┌───────┐
+    //   │   A   │
+    //   └──┬──┬─┘
+    //      │  └────────────┐
+    //      │           ┌───┴───┐
+    //      │post_op    │   B   │
+    //      │           └───┬───┘
+    //      │  ┌────────────┘
+    //   ┌──┴──┴─┐
+    //   │   C   │
+    //   └───────┘
```

Comment:
Nit: How about adding a graph after the change?

Reply:
This is actually not about graph transformation itself. It is about mapping post-operation to either `binary_add` or `sum`